File: 463. Island Perimeter.java
public class Solution {
    public int islandPerimeter(int[][] grid) {
        int perim = 0;
        
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[0].length; j++) {
                if (grid[i][j] == 1) {
                    perim+=4;
                    
                    if (i != 0 && grid[i-1][j] == 1) { // north
                        perim-=1;
                    }
                    
                    if (i != grid.length-1 && grid[i+1][j] == 1) { // south
                        perim-=1;
                    }
                    
                    if (j != 0 && grid[i][j-1] == 1) { // west
                        perim-=1;
                    }
                    
                    if (j != grid[0].length-1 && grid[i][j+1] == 1) { // east
                        perim-=1;
                    }
                    
                }
            }
        }
        
        return perim;
    }
}

File: 463. Island Perimeter.py
class Solution:
    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        count = 0
        
        for x in range(len(grid)):
            for y in range(len(grid[0])):
                if grid[x][y] == 1:
                    count += 4
                    
                    if x < len(grid) - 1 and grid[x + 1][y] == 1:
                        count -= 1
                    if y < len(grid[0]) - 1 and grid[x][y + 1] == 1:
                        count -= 1
                    if x > 0 and grid[x - 1][y] == 1:
                        count -= 1
                    if y > 0 and grid[x][y - 1] == 1:
                        count -= 1
                
        return count
